webcam.go

```diff
@@ -164,21 +175,34 @@ func (w *Webcam) StartStreaming() error {
 // If frame cannot be read at the moment
 // function will return empty slice
 func (w *Webcam) ReadFrame() ([]byte, error) {
+    result, index, err := w.GetFrame()
+    if err == nil {
+        w.ReleaseFrame(index)
+    }
+    return result, err
+}
+
+// Get a single frame from the webcam and return the frame and
+// the buffer index. To return the buffer, ReleaseFrame must be called.
+// If frame cannot be read at the moment
+// function will return empty slice
+func (w *Webcam) GetFrame() ([]byte, uint32, error) {
```

Comment:
Is there a use case for this?

Reply:
The primary case (and the main reason I suggested these changes) is to allow the grabbing of the most recent frames possible (i.e a small buffer count) and to allow some potentially lengthy image processing on them - I found that if the buffer was released back to the pool immediately, the camera could overwrite the buffer contents.

Reply:
ping?

Reply:
Sorry for late response. I'll finish the review today